scripts/screeps_tencent_batch_rl_runner.py

```
@@ -2266,6 +2275,188 @@ def verified_remote_candidate_scorecard(
+        raise BatchRunError("remote candidateScorecards.comparisons must be a list")
+    comparison_count = required_non_negative_int(raw.get("comparisonCount"), "candidateScorecards.comparisonCount")
+    if comparison_count != len(comparisons):
+        raise BatchRunError(
+            f"remote candidateScorecards.comparisonCount disagrees with comparisons: "
+            f"{comparison_count} != {len(comparisons)}"
+        )
+
+    verified_comparisons: list[dict[str, Any]] = []
+    for index, item in enumerate(comparisons):
+        if not isinstance(item, dict):
+            raise BatchRunError(f"remote candidateScorecards.comparisons[{index}] must be an object")
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)